Sources/MapboxMaps/Foundation/Extensions/Core/ResourceOptions.swift

```diff
@@ -63,4 +71,38 @@ extension ResourceOptions {
 
         return cacheDirectoryURL.appendingPathComponent("cache.db")
     }
+
+    /// :nodoc:
+    public override func isEqual(_ object: Any?) -> Bool {
+
+        guard let other = object as? ResourceOptions else {
+            return false
+        }
+
+        guard type(of: self) == type(of: other) else {
+            return false
+        }
+
+        return
+            (accessToken == other.accessToken) &&
+            (baseURL == other.baseURL) &&
+            (cachePath == other.cachePath) &&
+            (assetPath == other.assetPath) &&
+            (tileStorePath == other.tileStorePath) &&
+            (loadTilePacksFromNetwork == other.loadTilePacksFromNetwork) &&
+            (cacheSize == other.cacheSize)
+    }
+
+    /// :nodoc:
+    open override var hash: Int {
+        var hasher = Hasher()
+        accessToken.hash(into: &hasher)
+        baseURL?.hash(into: &hasher)
+        cachePath?.hash(into: &hasher)
+        assetPath?.hash(into: &hasher)
+        tileStorePath?.hash(into: &hasher)
+        loadTilePacksFromNetwork.hash(into: &hasher)
+        cacheSize?.hash(into: &hasher)
+        return hasher.finalize()
```

Comment:
I think you'd typically write this like…

```suggestion
        var hasher = Hasher()
        hasher.combine(accessToken)
        hasher.combine(baseURL)
        hasher.combine(cachePath)
        hasher.combine(assetPath)
        hasher.combine(tileStorePath)
        hasher.combine(loadTilePacksFromNetwork)
        hasher.combine(cacheSize)
        return hasher.finalize()
```